pubsub/gcp/gcp.go

```diff
@@ -163,7 +174,19 @@ func (p *publisher) PublishRaw(ctx context.Context, key string, m []byte) error
 		Data:       m,
 		Attributes: map[string]string{"key": key},
 	})
-	_, err := res.Get(ctx)
+	var err error
+	if p.topic.PublishSettings.DelayThreshold <= 1*time.Millisecond {
+		_, err = res.Get(ctx)
+	} else {
+		// if the DelayThreshold > 1 — the default — we use a goroutine
+		// otherwise we'll block here until that time interval passes
+		go func(res *gpubsub.PublishResult, ctx context.Context, m []byte) {
+			_, err := res.Get(ctx)
+			if err != nil {
```

Comment:
Let's inline this to make it clear it's not the `err` from the outside and scoped just to this part here.
`if _, err := res.Get(ctx); err != nil {`

Reply:
That's updated.